src/Utils/colecaoFirebase.ts

```diff
@@ -1,34 +1,450 @@
+      }
+      throw new Error(`${(error as Error).message}`);
+    }
+  };
+
+  /**
+   * atualiza uma coleção com os dados passados, caso a privacidade seja alterada
+   * a coleção é movida para o novo caminho em uma transacao e os itens são movidos
+   * para o novo caminho em uma outra transacao
+   * @param colecao
+   */
+  atualizarColecao = async (colecao: Omit<Colecao, "itens">): Promise<void> => {
+    if (!colecao.id) {
+      throw new Error("Id da coleção não informado");
+    }
+
+    const currentRef = doc(db, colecao.id).withConverter(this.#converter);
+
+    try {
+      const document = await getDoc(currentRef).catch((error) => {
+        throw new FirebaseError(error.code, "Erro ao buscar a coleção");
+      });
+      if (!document.exists()) {
+        throw new Error("Coleção não encontrada");
+      }
+      const colecaoData = document.data();
+      if (colecao.privado !== colecaoData.privado) {
+        const replaceableSubPath = colecao.id.includes("publico")
+          ? "publico"
+          : "privado";
+        const newSubPath = colecao.privado ? "privado" : "publico";
+        const newPath = colecao.id.replace(replaceableSubPath, newSubPath);
+        const newRef = doc(db, newPath).withConverter(this.#converter);
+        //primeira transacao move a colecao para o novo caminho
+        await runTransaction(db, async (transaction) => {
+          await this.#moveCollectionWithUpdate(colecao, newPath, transaction);
+        });
+        // segunda transacao remove cada item do caminho antigo e adiciona no novo
+        // essa transacao ocorre somente apos a primeira para evitar leitura de dados antigos da colecao alvo
+        await runTransaction(db, async (transaction) => {
+          await this.#moveCollectionItems(
+            colecaoData.privado,
+            currentRef,
+            newRef,
+            transaction
+          );
+        });
+      } else {
+        const sendingData = this.#converter.pick(
+          colecao,
+          "nome",
+          "descricao",
+          "privado"
+        );
+        updateDoc(currentRef, sendingData).catch((error) => {
+          throw new FirebaseError(error.code, "Erro ao atualizar a coleção");
+        });
+      }
+    } catch (error) {
+      if ((error as FirebaseError).code == "permission-denied") {
+        (error as Error).message =
+          "Você não tem permissão para atualizar coleções";
+      }
+      throw new Error(`${(error as Error).message}`);
+    }
+  };
+  /**
+   * remove uma coleção e, opcionalmente, todos os itens associados a ela,
+   * caso deleteitems seja false uma nova colecao deve ser informada para mover os itens.
+   * @param path da coleção a ser deletada
+   * @param deleteitems para remover todos os itens da coleção antes de deletá-la
+   * @param targetCollectionPath caminho da nova coleção para mover os itens
+   */
+  deletarColecao = async (
```

Comment:
![image](https://github.com/user-attachments/assets/cf481d15-51f2-420d-830b-60ec5756b678)
Erro ao remover e coleção e itens
![image](https://github.com/user-attachments/assets/2fe5f927-70a5-4b03-83ca-35e6eacf460e)
Erro ao deletar coleção e mover itens

Reply:
Como as coleções e os itens estavam salvos no firestore?

Reply:
![image](https://github.com/user-attachments/assets/c88dff1c-5334-44af-b98e-ff1c1e4c24ac)
![image](https://github.com/user-attachments/assets/a4a369e4-938c-4591-b98c-59096cb903da)

Reply:
Se o documento da coleção (publico/lista/algum_id) for vazio, o firestore não vai retornar ele, mesmo que o caminho exista, no método de adicionar uma coleção ela já é adicionada com os atributos existentes

Reply:
![image](https://github.com/user-attachments/assets/c9c83780-33a6-4b06-8e7e-52d6de4e70cf)
Documento não está vazio

Reply:
Mesmo colocando os atributos no documento da coleção o método de mover aponta o mesmo erro, mas o remover funciona após adicionar os atributos.